crates/katana/core/src/messaging/service.rs

```diff
@@ -0,0 +1,325 @@
+
+                pin.local_from_block += 1;
+                return Poll::Ready(Some(()));
+            }
+        }
+
+        // Poll the settling future.
+        if let Some(mut settling) = pin.settling.take() {
+            if let Poll::Ready(()) = settling.poll_unpin(cx) {
+                // +1 to move to the next local block to check messages to be
+                // sent on the settlement chain.
+                pin.local_from_block += 1;
+                return Poll::Ready(Some(()));
+            } else {
+                pin.settling = Some(settling)
+            }
+        }
+
+        Poll::Pending
+    }
+}
+
+/// Returns an `Interval` from the given seconds.
+fn interval_from_seconds(secs: u64) -> Interval {
+    let duration = Duration::from_secs(secs);
+    let mut interval = interval_at(Instant::now() + duration, duration);
+    interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+    interval
+}
+
+fn trace_msg_to_l1_sent(messages: &Vec<MsgToL1>, hashes: &Vec<String>) {
+    assert_eq!(messages.len(), hashes.len());
+    let hash_exec_str = format!("{:#064x}", HASH_EXEC);
+
+    for (i, m) in messages.iter().enumerate() {
+        let payload_str: Vec<String> = m.payload.iter().map(|f| format!("{:#x}", *f)).collect();
+
+        let hash = &hashes[i];
+
+        if hash == &hash_exec_str {
+            let to_address = &payload_str[0];
+            let selector = &payload_str[1];
+            let payload_str = &payload_str[2..];
+
+            info!(target: MSGING_TARGET,
+                                          r"Message executed on settlement layer:
+| from_address | {:#x}
+|  to_address  | {}
+|   selector   | {}
+|   payload    | [{}]
+
+",
+                                          m.from_address,
+                                          to_address,
+                                          selector,
+                                          payload_str.join(", ")
+                                    );
+        } else {
+            // We check for magic value 'MSG' used only when we are doing L3-L2 messaging.
+            let (to_address, payload_str) = if format!("{:#x}", m.to_address) == "0x4d5347" {
+                (payload_str[0].clone(), &payload_str[1..])
+            } else {
+                (format!("{:#x}", m.to_address), &payload_str[..])
+            };
+
+            info!(target: MSGING_TARGET,
+                                          r"Message sent to settlement layer:
+|     hash     | {}
+| from_address | {:#x}
+|  to_address  | {}
+|   payload    | [{}]
+
+",
+                                          hash.as_str(),
+                                          m.from_address,
+                                          to_address,
+                                          payload_str.join(", ")
+                                    );
+        }
+    }
+}
+
+fn trace_l1_handler_tx_exec(tx: &L1HandlerTransaction) {
```

Comment:
not sure if we want to have such verbose log, maybe just the tx hashes would suffice?

Reply:
The main reason is that, when debugging messaging I found it useful to have the content of the transaction (especially the payload) to debug my ethereum code to ensure I am serializing the felts correctly.

As for the other one printing the message hash + the content allows a developer to recompute the hash by hash is needed for debug with the content. But also helps going faster to test the message manual consuming as you do need the whole payload to consume the message.